### PR TITLE
Rails 4

### PR DESCRIPTION
**Pull request by naiyt**
These are minor updates to make our Doorkeeper fork Rails 4 compatible. At some point we should completely upgrade to the latest Doorkeeper, but that will take a bit more work.

### CODE REVIEW

**Comment by cantino** [comment]
Are these copied from the mainline gem, or written ourselves?

**Comment by naiyt** [comment]
Written ourselves. The mainline gem has many more changes, this is the minimum to get it working without deprecations on Rails 4.

**Comment by cantino** [comment]
Okay, let's merge after rails4 release.

File: lib/doorkeeper/rails/routes.rb [hunk 1]
```diff
@@ -40,10 +40,15 @@ def map_route(name, method)
       end
 
       def authorization_routes(mapping)
-        routes.scope :controller => mapping[:controllers] do
-          routes.match 'authorize', :via => :get,    :action => :new, :as => mapping[:as]
-          routes.match 'authorize', :via => :post,   :action => :create, :as => mapping[:as]
-          routes.match 'authorize', :via => :delete, :action => :destroy, :as => mapping[:as]
+        routes.resource(
```
Comment (cantino) [inline lib/doorkeeper/rails/routes.rb]: It seems like the exposed routes changed here a bunch.  Before it was get, post, and delete.  We've added update and removed new.  Have you click tested adding, updating, and removing oauth applications and authorizations?

Comment (cantino) [inline lib/doorkeeper/rails/routes.rb]: /cc @missvallord, @naiyt

Comment (naiyt) [inline lib/doorkeeper/rails/routes.rb]: Yep, all seems to work fine.

Comment (cantino) [inline lib/doorkeeper/rails/routes.rb]: Thanks!

If mobile / chrome are working, that also shows that most stuff here works.

**Pull request merged**